server/services/src/main/java/org/zanata/service/impl/MachineTranslationServiceImpl.java

```diff
@@ -209,7 +210,10 @@ private MTDocument getTranslationFromMT(MTDocument request,
             log.info("No documents in {}", version.userFriendlyToString());
             return AsyncTaskResult.completed();
         }
-        taskHandle.setMaxProgress(documents.size());
+        // Set taskHandle to count the textflows of all documents
+        taskHandle.setUnits(
+                documents.values().stream()
+                        .mapToLong(d -> d.getTextFlows().size()).sum());
```

Comment:
This could be very expensive (N+1 fetches) unless we write a custom query which returns all the project's text flows in one go (or better still, just the **number** of non-obsolete text flows in non-obsolete documents).